Tools/CISupport/ews-build/steps.py

```diff
@@ -2117,14 +2134,35 @@ def filter_logs_containing_error(self, logs, searchString='rror:', max_num_lines
     def send_email_for_new_build_failure(self):
         try:
             patch_id = self.getProperty('patch_id', '')
-            # FIXME: Support pull requests
-            if not patch_id or not self.should_send_email(patch_id):
+            pr_number = self.getProperty('github.number', '')
+            sha = self.getProperty('github.head.sha', '')[:HASH_LENGTH_TO_DISPLAY]
+
+            if patch_id and not self.should_send_email_for_patch(patch_id):
+                return
+            if pr_number and not self.should_send_email_for_pr(pr_number):
                 return
+
+            change_string = None
+            change_author = None
+            if patch_id:
+                change_author = self.getProperty('patch_author', '')
+                change_string = 'Patch {}'.format(patch_id)
+            elif pr_number and sha:
+                change_string = 'Commit {}'.format(sha)
+                change_author, errors = GitHub.email_for_owners(self.getProperty('owners', []))
+                for error in errors:
+                    print(error)
+            else:
+                print('No change detected, failed to determine author of change')
+                return
+            if not change_author:
+                print('Failed to determine change author')
```

Comment:
Better to print some more information which can help in debugging, e.g.: hash, pr_number etc.

Reply:
I believe this will happen frequently, when the email is not in contributors.json. It's probably better to show this in UI (using _addToLog), that we couldn't find the GitHub username in contributors.json and so skipping sending email notification.